tests/epics/adcore/test_writers.py

```diff
@@ -67,25 +81,36 @@ async def detectors(
     async with init_devices(mock=True):
         detectors.append(advimba.VimbaDetector("VIMBA:", static_path_provider))
         detectors.append(adkinetix.KinetixDetector("KINETIX:", static_path_provider))
-        detectors.append(adpilatus.PilatusDetector("PILATUS:", static_path_provider))
+        detectors.append(
+            adpilatus.PilatusDetector(
+                "PILATUS:", static_path_provider, PilatusReadoutTime.PILATUS3
+            )
+        )
         detectors.append(adaravis.AravisDetector("ADARAVIS:", static_path_provider))
     return detectors
 
 
-async def test_collect_stream_docs(hdf_writer: adcore.ADHDFWriter):
+async def test_hdf_writer_collect_stream_docs(hdf_writer: adcore.ADHDFWriter):
     assert hdf_writer._file is None
 
     [item async for item in hdf_writer.collect_stream_docs(1)]
     assert hdf_writer._file
 
 
+async def test_tiff_writer_collect_stream_docs(tiff_writer: adcore.ADTIFFWriter):
+    assert tiff_writer._emitted_resource is None
+    [item async for item in tiff_writer.collect_stream_docs(1)]
+    assert tiff_writer._emitted_resource
+
```

Comment:
Going off of my previous comment, shouldn't we call `tiff_writer.open()` prior to collecting the stream docs?
```suggestion
async def test_tiff_writer_collect_stream_docs(tiff_writer: adcore.ADTIFFWriter):
    assert tiff_writer._emitted_resource is None
    _ = await tiff_writer.open(1)
    [item async for item in tiff_writer.collect_stream_docs(1)]
    assert tiff_writer._emitted_resource

```